src/generate.py

```diff
@@ -32,7 +32,7 @@ def to_domain_ublock(url):
         return f"||{formated_url}^$all"
 
 def to_google(url):
-    return f'google.*##.g:has(a[href*="{format_url(url)}"])'
+    return f'google.*###rso .MjjYud a[href*="{format_url(url)}"]:upward(.MjjYud)'
```

Comment:
`MjjYud` is a minified class name, it could change on the next deploy. `#rso` is indeed stable though, and does encapsulate the search results only, which will allow to ignore the search bar.

Hiding the `g` should be enough for the purposes of this filter, but using `:upward(.g)` might look wonky on grouped results 
[such as these](https://user-images.githubusercontent.com/6241083/203869923-40851e34-3a89-4f1f-954a-ee36aa2f948c.png) (this layout has a toplevel `.g`, then per-results `.g`, so links will be removed, but some chrome might persist.

I think that the following should be enough. `:has` is indeed sub-optimal, but it's scoped to the search results container, and won't match on the autocomplete:
```suggestion
    return f'google.*###rso .g:has(a[href*="{format_url(url)}"])'
```

Reply:
Unfortunately, on mobile, there are no `.g`. 
`.MjjYud` was the best I got. Classes in Google css don't change that much. But you are right: it required more tests.

Reply:
Makes sense, I didn't think about mobile!